app/src/main/kotlin/at/bitfire/davdroid/push/PushNotificationManager.kt

```diff
@@ -0,0 +1,49 @@
+package at.bitfire.davdroid.push
+
+import android.accounts.Account
+import android.content.Context
+import androidx.core.app.NotificationCompat
+import at.bitfire.davdroid.R
+import at.bitfire.davdroid.ui.NotificationRegistry
+import java.util.logging.Logger
+
+object PushNotificationManager {
+    private val logger = Logger.getLogger("PushNotificationManager")
+
+    /**
+     * Generates the notification ID for a push notification.
+     */
+    private fun notificationId(account: Account, authority: String): Int {
+        return account.name.hashCode() + account.type.hashCode() + authority.hashCode()
+    }
+
+    /**
+     * Sends a notification to inform the user that a push notification has been received, the
+     * sync has been scheduled, but it still has not run.
+     */
+    fun notifyScheduled(context: Context, account: Account, authority: String) {
```

Comment:
Maybe the `Scheduled` postfix could be misleading. It sounds a bit like if the notification was scheduled (not the sync).

So I suggest just `notify` (and `dismiss`) because the context can be derived from the class name.